spec/javascripts/services/geocoding_service_spec.coffee

```diff
@@ -0,0 +1,78 @@
+do ->
+  'use strict'
+  describe 'GeocodingService', ->
+    GeocodingService = undefined
+    httpBackend = undefined
+    fakeShortFormDataService =
+      formatUserDOB: ->
+    fakeValidResult = getJSONFixture('geocoding-api-valid-address.json')
+    fakeInvalidResult = getJSONFixture('geocoding-api-invalid-address.json')
+    requestURL = undefined
+    fakeListing = getJSONFixture('listings-api-show.json')
+    fakeAddress =
+      address1: '4053 18th st'
+      city: 'San Francisco'
+      state: 'CA'
+      zip: '94114'
+    fakeApplicant =
+      firstName: 'Bob'
+      lastName: 'Williams'
+      dob_month: '07'
+      dob_day: '05'
+      dob_year: '2015'
+      preferences: {liveInSf: null, workInSf: null}
+    fakeHouseholdMember =
+      firstName: 'Ethel'
+      lastName: 'O\'Keefe'
+      dob_month: '09'
+      dob_day: '27'
+      dob_year: '1966'
+    geocodeOptions =
+      address: fakeAddress
+      listing: fakeListing
+      member: fakeHouseholdMember
+      applicant: fakeApplicant
+      adhp: true
+
+    beforeEach module('ui.router')
+    beforeEach module('dahlia.services', ($provide)->
+      $provide.value 'ShortFormDataService', fakeShortFormDataService
+      return
+    )
+
+    beforeEach inject((_GeocodingService_, _$httpBackend_) ->
+      httpBackend = _$httpBackend_
+      GeocodingService = _GeocodingService_
+      return
+    )
+
+    describe 'geocode', ->
```

Comment:
Note that the following tests will need to be updated slightly when both this PR and the PR for updating the boundary match values are merged into master, because that PR adjusts what values we should expect to be returned for the boundary match.